frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -818,7 +863,24 @@ func dispatchCopyFileOp(d *dispatchState, c instructions.SourcesAndDest, sourceS
 		}
 	}
 
-	commitMessage.WriteString(" " + c.Dest())
+	for _, src := range c.SourceContents {
+		data := src.Data
+		f := src.Path
+		st := llb.Scratch().Dir("/").File(llb.Mkfile(f, 0664, []byte(data)))
+
+		opts := append([]llb.CopyOption{&llb.CopyInfo{
```

Comment:
looks like this is missing user options(maybe more). If possible would be better to avoid duplication here compared to the `copy` code above that looks quite similar.

Reply:
Mm, I agree with the repetition point I think.

Not sure what you mean about user options? To get this behavior, I've kind of attempted to copy how HTTP files are added (although, they ignore the permissions, see #2114). Is it intentional that there are different sets of copy options for http files and local files - I can't see any options that are applied in either that should make a difference.

If they really should just all be the same, then it'll be nice to refactor all of this, and we can even fix #2114 along with it!

Reply:
> Not sure what you mean about user options?

Correct, I missed that this was part of `copyOpt` already.